public/app/features/provisioning/components/Shared/ProvisioningAwareFolderPicker.tsx

```diff
@@ -0,0 +1,59 @@
+import { skipToken } from '@reduxjs/toolkit/query';
+import { useMemo } from 'react';
+
+import { config } from '@grafana/runtime';
+import { useGetFrontendSettingsQuery } from 'app/api/clients/provisioning/v0alpha1';
+import { NestedFolderPickerProps } from 'app/core/components/NestedFolderPicker/NestedFolderPicker';
+import { FolderPicker } from 'app/core/components/Select/FolderPicker';
+
+import { useIsProvisionedInstance } from '../../hooks/useIsProvisionedInstance';
+
+interface Props extends NestedFolderPickerProps {
+  /* repo name (uid) or undefined (when it's non-provisioned folder)*/
+  repositoryName?: string;
+  isNonProvisionedFolder?: boolean;
+}
+
+export function ProvisioningAwareFolderPicker({ repositoryName, isNonProvisionedFolder, ...props }: Props) {
+  const isProvisionedInstance = useIsProvisionedInstance();
+  const provisioningEnabled = config.featureToggles.provisioning;
+  const { data: settingsData } = useGetFrontendSettingsQuery(provisioningEnabled ? undefined : skipToken);
+
+  const rootFolderUID = useMemo(() => {
```

Comment:
Just a hunch, but I think memoizing these ifs may be more costly than just running them on each render